docs/src/architecture/09_architecture_decisions/0010-frontend-json-schema-constraints.md

```diff
@@ -0,0 +1,170 @@
+<!-- markdownlint-enable max-one-sentence-per-line -->
+
+## Schema Evolution Guidelines
+
+To safely evolve the schema over time while preserving functionality:
+
+1. **Never rename predefined `nodeId`s** - If structure needs to change, keep the old nodeId as an alias.
+  The app must lookup both the new nodeId but fallback to old nodeId if new one not found.
+2. **Use versioning** - Include schema version in document metadata
+3. **Add new fields rather than changing existing ones** - Mark old fields as deprecated
+4. **Test template changes** - Automated testing should verify that all predefined nodeIds are still accessible
+
+## Risks and Mitigations
+
+| Risk | Impact | Mitigation |
+|------|--------|------------|
+| Template modifications break nodeId lookups | UI shows placeholders instead of actual content | Create automated tests that verify all required nodeIds exist in template updates |
+| Schema grows too complex with custom logic | Maintenance burden increases | Document all custom behaviors and create a clear process for adding new ones |
+| Template authors don't understand constraints | Templates created that don't work with the app | Create a template authoring guide with clear examples and validation tools |
+| Performance degrades with complex schemas | App becomes slow when rendering large documents | Implement performance testing for templates and optimize lookup algorithms |
+
+## Testing Strategy
+
+To ensure the integrity of our approach:
+
+1. **Unit Tests** - Verify nodeId resolution logic works correctly.
+2. **Integration Tests** - Check that UI components correctly display data from schema nodes.
+3. **Schema Validation** - Create validators that check for required nodeIds in templates.
+4. **Migration Tests** - Verify that document migrations preserve required fields.
+5. **Error Handling Tests** - Verify that the system gracefully handles missing or malformed nodes.
+6. **Performance Tests** (*Optional*) - Measure rendering time with different schema complexities.
+
+## Maintenance Plan
+
+### Current Issues
+
+The frontend team currently spends an excessive amount of time reviewing schema change PRs from architects
+who often make changes without validating against frontend constraints.
+This creates an unsustainable workflow where:
+
+1. Contributors create schema changes without understanding frontend requirements
+2. PRs reach review stage with fundamental issues that break nodeId lookups
+3. Frontend team must thoroughly review every schema change
+4. Cycles of revisions delay implementation and waste development resources
+
+### Improved Process
+
+To address these issues, we suggest implementing the following improved maintenance process:
+
+#### Automated Validation
+
+* **CI/CD Integration**: Add an automated check in the CI pipeline that fails PRs with breaking schema changes
```

Comment:
Ultimately ALL proposal templates will be sourced from the front end.  This is a temporary bootstrapping problem ONLY.